src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,54 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final float LOAD_COEFFICIENT = 1.5f;
+    private T[] stack;
+    private int size;
+    private int capacity;
 
+    public MateStack() {
+        size = 0;
+        capacity = 10;
+        stack = (T[]) new Object[capacity];
+    }
+
+    public void push(T value) {
+        if (size > capacity) {
+            grow();
+        }
+        stack[size] = value;
+        size++;
     }
 
     public T peek() {
-        return null;
+        return getLastItem();
     }
 
     public T pop() {
-        return null;
+        T lastItem = getLastItem();
+        size--;
```

Comment:
it will also be good to make the element on `stack[size - 1]` null before we decrement `size`